File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {

        
        char[] sArr = s.toCharArray();
        char[] tArr = t.toCharArray();
        int a = sArr.length;
        int b = tArr.length;
        
        int i=0, j=0;
        while (i!=a && j!=b) {
            if (sArr[i] == tArr[j]) {
                i++;
            }
            j++;
        }
        if (i == a) {
            return true;
        }
        else {
            return false;
        }
        
    }
}

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        if not s:
            return True

        i = 0
        for char in t:
            if s[i] == char:
                i += 1
                if i == len(s):
                    return True
        return False